R/Dataset.R

```diff
@@ -25,12 +28,15 @@ print.Dataset <- function(x, ...) {
   invisible(x)
 }
 
-`[.Dataset` <- function(x, i, ...) {
+`[.Dataset` <- function(x, i, inplace = FALSE, ...) {
   if (!missing(..1)) {
     if(!x$target %in% ...)
       stop("Cannot remove target column ", deparse(x$target))
   }
   dimensions <- dim(x$env$data)
+  if (inplace == FALSE) {
+    return(x$env$data[i, ...])
```

Comment:
This returns a data.frame instance instead of a Dataset.
I think we should return a Dataset instance here instead.